subsys/modbus/Kconfig

```diff
@@ -47,6 +47,29 @@ config MODBUS_SERIAL
 	help
 	  Enable Modbus over serial line support.
 
+choice
+
+	prompt "Supported serial drivers"
+	default MODBUS_SERIAL_IRQ
+	help
+	  Specify which serial driver to be used.
+
+config MODBUS_SERIAL_IRQ
+	bool "Modbus over serial line using IRQ uart driver"
+	depends on SERIAL && SERIAL_HAS_DRIVER
+	depends on DT_HAS_ZEPHYR_MODBUS_SERIAL_ENABLED
+	help
+	  Enable Modbus over serial line support using IRQ method.
+
+config MODBUS_SERIAL_ASYNC
+	bool "Modbus over serial line using Async uart driver"
+	depends on SERIAL && SERIAL_HAS_DRIVER && UART_ASYNC_API
+	depends on DT_HAS_ZEPHYR_MODBUS_SERIAL_ENABLED
+	help
+	  Enable Modbus over serial line support using ASYNC method.
+
+endchoice
+
```

Comment:
NACK to new Kconfig options, I do not see the need for them.

Reply:
I would like to understand this point better : 
Let's imagine the case where we have 2 UART peripherals.  
Only one use the Modbus protocol on top of it.  
The idea I had is that the regular UART could use async while we would still like to have the second UART with Modbus using interrupts or vice versa. Without this addition I guess both UART would be force to use the same API. 
Am I mistaking ?

Reply:
From modbus perspective it does not matter how another UART interface is configure or used. If it has advantages to use ASYNC API (looks like only RTU would make sense), then it should be preferred when an UART driver supports it, if not, modbus serial support should fall back to IRQ API, always, no options, simple code.